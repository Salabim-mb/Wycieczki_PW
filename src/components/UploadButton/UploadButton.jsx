import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '@material-ui/core';
import { Alert } from '@material-ui/lab';

const UploadButton = ({ header, icon, handleChange, error, ...rest }) => (
	<>
		<Button variant="contained" component="label" color="default" startIcon={icon}>
			{header}
			<input name={rest.id} type="file" {...rest} hidden onChange={handleChange} />
		</Button>
		{error && <Alert severity="error">{error}</Alert>}
	</>
);

UploadButton.propTypes = {
	header: PropTypes.string.isRequired,
	// eslint-disable-next-line react/forbid-prop-types
	icon: PropTypes.object.isRequired,
	handleChange: PropTypes.func.isRequired,
	error: PropTypes.string,
};

UploadButton.defaultProps = {
	error: '',
};

export default UploadButton;
