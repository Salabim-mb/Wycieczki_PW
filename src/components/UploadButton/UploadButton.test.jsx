import React from 'react';
import { render } from '@testing-library/react';
import UploadButton from './UploadButton';

describe('UploadButton', () => {
	const props = {
		header: 'Dodaj plik',
		icon: <>xd</>,
		handleChange: jest.fn(),
		error: '',
	};

	it('should match snapshot', () => {
		const { container } = render(<UploadButton {...props} />);

		expect(container).toMatchSnapshot();
	});

	it('should display error', () => {
		props.error = 'Błąd';
		const { container } = render(<UploadButton {...props} />);

		expect(container).toMatchSnapshot();
	});
});
