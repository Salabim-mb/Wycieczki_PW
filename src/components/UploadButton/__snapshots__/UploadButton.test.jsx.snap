// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadButton should display error 1`] = `
<div>
  <label
    aria-disabled="false"
    class="MuiButtonBase-root MuiButton-root MuiButton-contained"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiButton-label"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        xd
      </span>
      Dodaj plik
      <input
        hidden=""
        type="file"
      />
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </label>
  <div
    class="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
    role="alert"
  >
    <div
      class="MuiAlert-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      class="MuiAlert-message"
    >
      Błąd
    </div>
  </div>
</div>
`;

exports[`UploadButton should match snapshot 1`] = `
<div>
  <label
    aria-disabled="false"
    class="MuiButtonBase-root MuiButton-root MuiButton-contained"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiButton-label"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        xd
      </span>
      Dodaj plik
      <input
        hidden=""
        type="file"
      />
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </label>
  
</div>
`;
